addons/account/models/account_move.py

```diff
@@ -2045,7 +2045,7 @@ def post(self):
             if not move.line_ids.filtered(lambda line: not line.display_type):
                 raise UserError(_('You need to add a line before posting.'))
             if move.auto_post and move.date > fields.Date.today():
-                date_msg = move.date.strftime(self.env['res.lang']._lang_get(self.env.user.lang).date_format)
+                date_msg = move.date.strftime(self.env['res.lang']._lang_get(self.env.user.lang).date_format or "%m/%d/%Y")
```

Comment:
```suggestion
                date_msg = move.date.strftime(self.env['res.lang']._lang_get(self.env.user.lang).date_format or 'en_US')
```

Reply:
en_US might not be installed by default. Check `get_lang` instead:

https://github.com/odoo/odoo/blob/35a0afaa5aaa57c9b8d0c764c92431a34f80802f/odoo/tools/misc.py#L1166

Reply:
updated with get_lang, that's much better